src/darwin/Framework/CHIP/templates/CHIPCallbackBridge_internal.zapt

```diff
@@ -62,6 +72,10 @@ typedef void (*Nullable{{asUpperCamelCase parent.name}}Cluster{{asUpperCamelCase
 {{#chip_server_cluster_attributes}}
 {{#if isArray}}
 {{#>CHIPCallbackBridge header="1" ns=parent.name                                }}{{asUpperCamelCase ../../name}}{{asUpperCamelCase ../name}}ListAttributeCallback{{/CHIPCallbackBridge}}
+{{else}}
+  {{#if_is_struct type}}
+  {{#>CHIPCallbackBridge header="1" ns=parent.name                              }}{{asUpperCamelCase ../../name}}{{asUpperCamelCase ../name}}StructAttributeCallback{{/CHIPCallbackBridge}}
+  {{/if_is_struct}}
```

Comment:
Maybe you could have do the `if_is_struct` first. That would have avoided one level of nesting ?

Reply:
Unfortunately a list-of-structs attribute would then end up with the wrong thing here, because it tests true for both isArray and `if_is_struct type`, right?

Reply:
Oh right :(